src/arch/x86_64/mm/physicalmem.rs

```diff
@@ -19,6 +19,47 @@ const KVM_32BIT_MAX_MEM_SIZE: usize = 1 << 32;
 const KVM_32BIT_GAP_SIZE: usize = 768 << 20;
 const KVM_32BIT_GAP_START: usize = KVM_32BIT_MAX_MEM_SIZE - KVM_32BIT_GAP_SIZE;
 
+fn detect_from_fdt() -> Result<(), ()> {
```

Comment:
Since this is fallible (returns `Result`), this should also be used.

Ideally, things that can reasonably fail should return an error instead of panicking. Then, the next method for detection is used.
For example, we can use the `Option` from `get_fdt` with
```rust
let fdt = get_fdt().ok_or(())?;
```